src/frontend/src/components/FilterPanel/components/LineageFilter/index.tsx

```diff
@@ -1,32 +1,48 @@
-import { ComplexFilter } from "czifui";
+import {
+  ComplexFilter,
+  ComplexFilterValue,
+  DefaultMenuSelectOption,
+} from "czifui";
 import React, { useEffect, useState } from "react";
-import { DefaultMenuSelectOption } from "../../index";
 
 interface Props {
   options: DefaultMenuSelectOption[];
   updateLineageFilter: (lineages: string[]) => void;
 }
 
+// ComplexFilter defaults to checking if option is selected (is a value) by equality.
+// However, because our options are objects that are dynamically generated, object
+// equality is always false because they're not the same object, even if same content.
+// Instead, we must create a custom selection checker to check underlying value.
+const getOptionSelected = (
+  option: DefaultMenuSelectOption,
+  value: DefaultMenuSelectOption
+) => {
+  return option.name === value.name;
+};
+// ComplexFilter doesn't directly do the check, it's done by its child MenuSelect
+const optionCheckingMenuSelectProps = {
```

Comment:
nit: I was thinking the variable name could possibly be just `MenuSelectProps`, since that's what the object is, and also means if we add more props to the object we don't need to rename the variable from `optionCheckingMenuSelectProps` to `optionCheckingAndFooMenuSelectProps`, etc..

What do you think?

Reply:
So I generally do that kind of naming, here's why I didn't do it here (although I actually was waffling on do it):

- I wanted to be explicit about the purpose of passing these props. They're just there to target one aspect -- the option equality checking -- and I wanted that to be super clear. But your point is well taken: if we ever had to go back and add more MenuSelectProps, it would necessitate either changing the name to something absurdly specific or just generalizing it back to MenuSelectProps. Really, if someone is going to go changing it, they'll go read where it's defined and see what's up, so yeah, I've fully come around to your POV on this aspect.
- The other thing, and it's a little silly, but it gave me a way to avoid PascalCase naming. Since it's an object, I wanted to keep it in camelCase, and by tacking some words on the front explaining what the props were doing, it gave me a way to have my cake and eat it too.

Since I no longer agree with my first point after thinking about what you said, I'm on board with naming it `MenuSelectProps`, but I do kind of miss keeping it camelCase since it's an object.